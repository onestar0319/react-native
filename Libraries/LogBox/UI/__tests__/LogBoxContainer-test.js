/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow
 */

'use strict';

const React = require('react');
const LogBoxContainer = require('../LogBoxContainer').default;
const LogBoxLog = require('../../Data/LogBoxLog').default;
const render = require('../../../../jest/renderer');

describe('LogBoxContainer', () => {
  it('should render null with no logs', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={new Set()}
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render the latest warning', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'warn',
              {
                content: 'Some kind of message',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
            new LogBoxLog(
              'warn',
              {
                content: 'Some kind of message (latest)',
                substitutions: [],
              },
              [],
              'Some kind of message (latest)',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render the latest error', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'error',
              {
                content: 'Some kind of message',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
            new LogBoxLog(
              'error',
              {
                content: 'Some kind of message (latest)',
                substitutions: [],
              },
              [],
              'Some kind of message (latest)',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render both an error and warning', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'warn',
              {
                content: 'Some kind of message',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
            new LogBoxLog(
              'error',
              {
                content: 'Some kind of message (latest)',
                substitutions: [],
              },
              [],
              'Some kind of message (latest)',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render null when disabled with non-fatal error and warning', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        isDisabled
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'warn',
              {
                content: 'Some kind of message',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
            new LogBoxLog(
              'error',
              {
                content: 'Some kind of message (latest)',
                substitutions: [],
              },
              [],
              'Some kind of message (latest)',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render fatal before warn and error', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'fatal',
              {
                content: 'Some kind of fatal message',
                substitutions: [],
              },
              [],
              'Some kind of fatal message',
              [],
            ),
            new LogBoxLog(
              'warn',
              {
                content: 'Some kind of message',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
            new LogBoxLog(
              'error',
              {
                content: 'Some kind of message (latest)',
                substitutions: [],
              },
              [],
              'Some kind of message (latest)',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render most recent fatal', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'fatal',
              {
                content: 'Should not be selected',
                substitutions: [],
              },
              [],
              'Some kind of fatal message',
              [],
            ),
            new LogBoxLog(
              'fatal',
              {
                content: 'Should be selected',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render fatal error even when disabled', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        isDisabled
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'fatal',
              {
                content: 'Should be selected',
                substitutions: [],
              },
              [],
              'Some kind of message',
              [],
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render most recent syntax error', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'syntax',
              {
                content: 'Should not be selected',
                substitutions: [],
              },
              [],
              'Some kind of syntax error message',
              [],
              {
                fileName:
                  '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
                location: {row: 199, column: 0},
                content: `  197 | });
  198 |
> 199 | export default CrashReactApp;
      | ^
  200 |`,
              },
            ),
            new LogBoxLog(
              'syntax',
              {
                content: 'Should be selected',
                substitutions: [],
              },
              [],
              'Some kind of syntax error message',
              [],
              {
                fileName:
                  '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
                location: {row: 199, column: 0},
                content: `  197 | });
  198 |
> 199 | export default CrashReactApp;
      | ^
  200 |`,
              },
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render syntax error even when disabled', () => {
    const output = render.shallowRender(
      <LogBoxContainer
        isDisabled
        onDismiss={() => {}}
        onDismissWarns={() => {}}
        onDismissErrors={() => {}}
        logs={
          new Set([
            new LogBoxLog(
              'syntax',
              {
                content: 'Should be selected',
                substitutions: [],
              },
              [],
              'Some kind of syntax error message',
              [],
              {
                fileName:
                  '/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js',
                location: {row: 199, column: 0},
                content: `  197 | });
  198 |
> 199 | export default CrashReactApp;
      | ^
  200 |`,
              },
            ),
          ])
        }
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
