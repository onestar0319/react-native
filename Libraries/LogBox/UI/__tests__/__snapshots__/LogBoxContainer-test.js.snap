// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxContainer should render both an error and warning notification 1`] = `
<View
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="warn"
      log={
        LogBoxLog {
          "category": "Some kind of message",
          "codeFrame": undefined,
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "warn",
          "message": Object {
            "content": "Some kind of message",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={1}
    />
  </View>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="error"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "error",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={1}
    />
  </View>
</View>
`;

exports[`LogBoxContainer should render null with no logs 1`] = `null`;

exports[`LogBoxContainer should render null with no selected log and disabled 1`] = `null`;

exports[`LogBoxContainer should render selected fatal error even when disabled 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <LogBoxInspector
    logs={
      Array [
        LogBoxLog {
          "category": "Some kind of message",
          "codeFrame": undefined,
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "fatal",
          "message": Object {
            "content": "Should be selected",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        },
      ]
    }
    onChangeSelectedIndex={[Function]}
    onDismiss={[Function]}
    onMinimize={[Function]}
    selectedIndex={0}
  />
</View>
`;

exports[`LogBoxContainer should render selected syntax error even when disabled 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <LogBoxInspector
    logs={
      Array [
        LogBoxLog {
          "category": "Some kind of syntax error message",
          "codeFrame": Object {
            "content": "  197 | });
  198 |
> 199 | export default CrashReactApp;
      | ^
  200 |",
            "fileName": "/path/to/RKJSModules/Apps/CrashReact/CrashReactApp.js",
            "location": Object {
              "column": 0,
              "row": 199,
            },
          },
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "syntax",
          "message": Object {
            "content": "Should be selected",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        },
      ]
    }
    onChangeSelectedIndex={[Function]}
    onDismiss={[Function]}
    onMinimize={[Function]}
    selectedIndex={0}
  />
</View>
`;

exports[`LogBoxContainer should render the latest error notification 1`] = `
<View
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="error"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "error",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={2}
    />
  </View>
</View>
`;

exports[`LogBoxContainer should render the latest warning notification 1`] = `
<View
  style={
    Object {
      "bottom": 20,
      "left": 10,
      "position": "absolute",
      "right": 10,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 8,
        "marginBottom": 5,
        "overflow": "hidden",
      }
    }
  >
    <LogBoxLogNotification
      level="warn"
      log={
        LogBoxLog {
          "category": "Some kind of message (latest)",
          "codeFrame": undefined,
          "componentStack": Array [],
          "count": 1,
          "isComponentError": false,
          "level": "warn",
          "message": Object {
            "content": "Some kind of message (latest)",
            "substitutions": Array [],
          },
          "stack": Array [],
          "symbolicated": Object {
            "error": null,
            "stack": null,
            "status": "NONE",
          },
        }
      }
      onPressDismiss={[Function]}
      onPressOpen={[Function]}
      totalLogCount={2}
    />
  </View>
</View>
`;
